modules/core/src/main/scala/Schema.scala

```diff
@@ -242,6 +244,26 @@ object Schema {
     case class Sequence[A](value: Schema[A]) extends Schema[List[A]]
     case class Record[R](value: FreeApplicative[Field[R, *], R])
         extends Schema[R]
+
+    object Record {
+
+      // todo: law tests (?)
```

Comment:
It'd probably be possible (generating `Field` seems doable)...

Reply:
I lost any remnants of sanity I might've had and came up with this:

```scala
package dynosaur

import munit.DisciplineSuite
import org.scalacheck.Arbitrary
import org.scalacheck.Gen
import cats.kernel.Eq
import cats.free.FreeApplicative
import cats.laws.discipline._
import cats.laws.discipline.arbitrary._
import cats.implicits._

class LawSuite extends DisciplineSuite {
  implicit def arbitraryField[A: Schema](implicit
      arbFun: Arbitrary[Option[A] => Option[A]]
  ): Arbitrary[Schema.structure.Field[Option[A], Option[A]]] = Arbitrary {
    Gen.oneOf(
      Gen.resultOf(
        Schema.structure.Field.Required.apply[Option[A], Option[A]](
          _,
          Schema[A].xmap(_.some.asRight)(_.toRight(Schema.WriteError("none"))),
          _
        )
      ),
      Gen.resultOf(
        Schema.structure.Field.Optional.apply[Option[A], A](_, Schema[A], _)
      )
    )
  }

  implicit def arbitraryFreeApplicativeField[A: Arbitrary](implicit
      arbAA: Arbitrary[A => A],
      arbitraryField: Arbitrary[Schema.structure.Field[A, A]]
  ): Arbitrary[FreeApplicative[Schema.structure.Field[A, *], A]] = {

    def self(
        levels: Int
    ): Gen[FreeApplicative[Schema.structure.Field[A, *], A]] = {
      if (levels > 0) {
        Gen.oneOf(
          arbitraryField.arbitrary.map(FreeApplicative.lift(_)),
          Gen.resultOf(
            FreeApplicative.pure[Schema.structure.Field[A, *], A](_)
          ),
          for {
            fields <- self(levels - 1)
            f <- arbAA.arbitrary
          } yield fields.map(f)
        )
      } else {
        Gen.oneOf(
          arbitraryField.arbitrary.map(FreeApplicative.lift(_)),
          Gen.resultOf(FreeApplicative.pure[Schema.structure.Field[A, *], A](_))
        )
      }
    }

    Arbitrary(self(2))
  }

  implicit def arbitraryRecord[A](implicit
      arbFields: Arbitrary[FreeApplicative[Schema.structure.Field[A, *], A]]
  ): Arbitrary[Schema.Record[A]] =
    Arbitrary(Gen.resultOf(Schema.structure.Record.apply[A](_)))

  //cheating a bit here - we're not doing proper equality checks on any of the schemas in the fields or even the functions
  implicit def eqFieldOfAnything[A]: Eq[Schema.structure.Field[A, A]] =
    Eq.fromUniversalEquals

  implicit val eqWriteError: Eq[Schema.WriteError] = Eq.fromUniversalEquals
  implicit val eqDynamoValue: Eq[DynamoValue] = Eq.fromUniversalEquals

  implicit def eqByRecord[A: ExhaustiveCheck]
      : Eq[FreeApplicative[Schema.structure.Field[A, *], A]] = Eq.by { field =>
    ExhaustiveCheck[A].allValues.map {
      Schema.fields(field).write
    }
  }

  implicit def eqRecord[A](implicit
      eqFields: Eq[FreeApplicative[Schema.structure.Field[A, *], A]]
  ): Eq[Schema.Record[A]] = Eq.by(_.value)

  implicit val miniIntSchema: Schema[MiniInt] = Schema.int.imapErr(
    MiniInt.fromInt(_).toRight(Schema.ReadError("invalid value"))
  )(_.toInt)

  checkAll(
    "InvariantMonoidal[Schema.Record]",
    InvariantMonoidalTests[Schema.Record]
      .invariantMonoidal[Option[MiniInt], Option[MiniInt], Option[MiniInt]]
  )
}
```

I'm not happy 😅 it passes, but seems like too much for a single law test. Perhaps we just need to leave that instance without the law tests.

tbh it seems like something that could be generalized to any `FreeApplicative` where the algebra is a contravariant functor...